openmdao/core/component.py

```diff
@@ -484,7 +484,12 @@ def add_input(self, name, val=1.0, shape=None, src_indices=None, flat_src_indice
         }
 
         if src_indices is not None:
-            metadata['src_indices'] = np.asarray(src_indices, dtype=INT_DTYPE)
+
+            contains_slice = _is_slice(src_indices)
+            if contains_slice:
```

Comment:
No need to create a temporary variable here.  Just do  `if _is_slice(src_indices):`